src/facade/dragonfly_connection.cc

```diff
@@ -190,35 +191,35 @@ size_t Connection::PipelineMessage::StorageCapacity() const {
   return storage.capacity() + args.capacity();
 }
 
-template <class... Ts> struct Overloaded : Ts... {
-  using Ts::operator()...;
-
-  template <typename T, typename D> size_t operator()(const unique_ptr<T, D>& ptr) {
-    return operator()(*ptr.get());
-  }
-};
-
-template <class... Ts> Overloaded(Ts...) -> Overloaded<Ts...>;
-
 size_t Connection::MessageHandle::UsedMemory() const {
-  // TODO: don't count inline size
```

Comment:
I know it's a draft, but just in case: better separate all the refactoring and the functionality change into separate commits